src/decrypt.c

```diff
@@ -1,29 +1,36 @@
 #include "ece/keys.h"
 #include "ece/trailer.h"
 
+#include <assert.h>
 #include <string.h>
 
 #include <openssl/evp.h>
 #include <openssl/rand.h>
 
-typedef size_t (*max_decrypted_length_t)(uint32_t rs, size_t ciphertextLen);
+typedef int (*unpad_t)(uint8_t* block, bool lastRecord, size_t* blockLen);
 
-typedef int (*unpad_t)(uint8_t* block, bool isLastRecord, size_t* blockLen);
-
-// Returns the maximum decrypted length of an "aes128gcm" ciphertext.
+// Calculates the maximum plaintext length, including room for the padding
+// delimiter and padding.
 static inline size_t
-ece_aes128gcm_max_decrypted_length(uint32_t rs, size_t ciphertextLen) {
-  size_t numRecords = (ciphertextLen / rs) + 1;
+ece_plaintext_max_length(uint32_t rs, size_t padSize, size_t ciphertextLen) {
+  size_t overhead = padSize + ECE_TAG_LENGTH;
```

Comment:
This shouldn't overflow `size_t`. `overhead` will be either 17 (aes128gcm) or 18 (aesgcm), and `padSize` is only used internally. Added assertions for this.